packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -1044,6 +1046,10 @@ class CupertinoSliverNavigationBar extends StatefulWidget {
   /// Defaults to [NavigationBarBottomMode.automatic] if this is null and a [bottom] is provided.
   final NavigationBarBottomMode? bottomMode;
 
+  /// Callback called when the search field in [CupertinoSliverNavigationBar.search]
+  /// is focused or unfocused.
+  final ValueChanged<bool>? onSearchFocused;
```

Comment:
For the final version, we probably don't want to refer to it as `onSearchFocused`. Focused suggests it triggers when a keyboard navigates to it, which is probably not what we want. Maybe `onSearchActive`?